lib/ain-evm/src/block.rs

```diff
@@ -80,30 +80,30 @@ impl BlockService {
         match parent_gas_used.cmp(&parent_gas_target) {
             Ordering::Equal => Ok(parent_base_fee),
             Ordering::Greater => {
-                let gas_used_delta = parent_gas_used - parent_gas_target;
+                let gas_used_delta = parent_gas_used - parent_gas_target; // sub is safe due to cmp
                 let base_fee_per_gas_delta = self.get_base_fee_per_gas_delta(
                     gas_used_delta,
                     parent_gas_target,
                     parent_base_fee,
                     base_fee_max_change_denominator,
                 )?;
-                Ok(max(
-                    parent_base_fee + base_fee_per_gas_delta,
-                    initial_base_fee,
-                ))
+                Ok(match parent_base_fee.checked_add(base_fee_per_gas_delta) {
```

Comment:
Can do a saturating_add, but do we really have to? Why don't we just return error. Simplifies it and makes sure nothing on the chain can result in overflow/underflows.

Reply:
Throwing an error would mean that block production would stop for a while until a sufficiently full/empty block is mined. Doing the `saturating_add` would allow the block production to remain unaffected while respecting U256 bounds.

Reply:
We're at the edge case where base fee jumps so high to overflow these numbers. If we're at the point, we likely will have to dealing with economical decision well before. I'd suggest just error-ing out and let things cool down.

Reply:
Reason being: whatever we do post this stage, error-ing out will make for better options, than saturating that will change fee characteristic.

Reply:
Max base fee is a good alternative for now. Looks good.